staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java

```diff
@@ -147,35 +146,26 @@ public List<String> saveFiles(final TenantId tenantId, BatchId batchId, FileItem
         final List<String> fileNames = new ArrayList<>();
         final Map<String, String> binaryFilesUploaded = new HashMap<>();
         try (final SubBatchWriter subBatchWriter = new SubBatchWriter(inProgressBatchFolderPath.toFile(), subbatchSize)) {
-            while (true) {
 
-                final FileItemStream fileItemStream;
-                try {
-                    LOGGER.debug("Retrieving next part...");
-                    if (!fileItemIterator.hasNext()) {
-                        LOGGER.debug("No further parts.");
-                        break;
-                    }
-                    fileItemStream = fileItemIterator.next();
-                } catch (FileUploadException | IOException ex) {
-                    throw new IncompleteBatchException(ex);
-                }
+            LOGGER.debug("Retrieving next part...");
+
+            for (final Part part : parts) {
 
-                if (!fileItemStream.isFormField()) {
+                if (part.getSubmittedFileName() != null) {
```

Comment:
I think `part.getSubmittedFileName() != null` is equivalent to `!fileItemStream.isFormField()`